src/commands/registerCommands.ts

```diff
@@ -0,0 +1,87 @@
+import * as path from 'path';
+import { commands, ExtensionContext, Position, Uri, window, workspace } from "vscode";
+import { CancellationToken, ExecuteCommandParams, ExecuteCommandRequest, LanguageClient, ReferencesRequest, TextDocumentIdentifier } from "vscode-languageclient";
+import { markdownPreviewProvider } from "../markdownPreviewProvider";
+import { CommandConstants } from "./commandConstants";
+
+/**
+ * Register the commands for vscode-xml
+ *
+ * @param context the extension context
+ */
+export async function registerCommands(context: ExtensionContext, languageClient: LanguageClient) {
+
+  // XML Documentation commands
+  context.subscriptions.push(markdownPreviewProvider);
+  context.subscriptions.push(commands.registerCommand(CommandConstants.OPEN_DOCS_HOME, async () => {
+    const uri = 'README.md';
+    const title = 'XML Documentation';
+    const sectionId = '';
+    markdownPreviewProvider.show(context.asAbsolutePath(path.join('docs', uri)), title, sectionId, context);
+  }));
+  context.subscriptions.push(commands.registerCommand(CommandConstants.OPEN_DOCS, async (params: { page: string, section: string }) => {
+    const page = params.page.endsWith('.md') ? params.page.substr(0, params.page.length - 3) : params.page;
+    const uri = page + '.md';
+    const sectionId = params.section || '';
+    const title = 'XML ' + page;
+    markdownPreviewProvider.show(context.asAbsolutePath(path.join('docs', uri)), title, sectionId, context);
+  }));
+
+  // Code Lens actions
+  context.subscriptions.push(commands.registerCommand(CommandConstants.SHOW_REFERENCES, (uriString: string, position: Position) => {
```

Comment:
Create a new function registerCodeLensCommands and move the codeLens code to this function